src/parquet/column/properties.h

```diff
@@ -85,7 +85,16 @@ static constexpr ParquetVersion::type DEFAULT_WRITER_VERSION =
     ParquetVersion::PARQUET_1_0;
 static constexpr Compression::type DEFAULT_COMPRESSION_TYPE = Compression::UNCOMPRESSED;
 
-using ColumnCodecs = std::unordered_map<std::string, Compression::type>;
+class PARQUET_EXPORT ColumnSettings {
+ public:
+  ColumnSettings(Encoding::type encoding = DEFAULT_ENCODING,
+      Compression::type codec = DEFAULT_COMPRESSION_TYPE, bool collect_statistics = false)
+      : encoding(encoding), codec(codec), collect_statistics(collect_statistics) {}
+
+  Encoding::type encoding;
+  Compression::type codec;
+  bool collect_statistics;
+};
```

Comment:
If this is strictly POD, this should be a struct (note google style guidelines on member variable names for structs vs classes)